Fehler beim kompilieren

Hallo Sebastian,
habe 3 Soyos 1200W Typen, ohne eingebautes Display, mit Deiner Steuerung.
Funktionieren soweit gut.
Du hattest mir schon mehrfach geholfen die WR mit einem Tasmota Hitch IR Lesekopf zum
laufen zu bringen.
Nun wollte ich noch einen vierten in Betrieb nehmen.
Leider gibts beim kompilieren eine Fehlermeldung.
Das dafür genutzte YAML File ist bis auf die Namensgebung für den WR exakt das gleiche wie es auf den 3 anderen Soyos läuft.
Soll heissen: Vor ein paar Monaten konnte ich die Steuerungen, bauen, die Software ohne Fehler kompilieren, und all das in Betrieb nehmen.
Das geht jetzt leider nicht mehr, obwohl die YAML`s, bis auf die Namen nicht geändert wurden.
Hier das Fehler Log File:

INFO ESPHome 2026.4.5
INFO Reading configuration /config/esphome/soyo-4.yaml...
INFO Updating https://github.com/syssi/esphome-soyosource-gtn-virtual-meter.git@main
WARNING register_action('http_request.get', ...) is missing the synchronous= parameter. Defaulting to synchronous=False (safe but prevents StringRef optimization). Check the C++ class: use synchronous=False if play_next_() is deferred to a callback, timer, or loop(); use synchronous=True if play_next_() always runs before the initial play/play_complex call returns
WARNING register_action('http_request.post', ...) is missing the synchronous= parameter. Defaulting to synchronous=False (safe but prevents StringRef optimization). Check the C++ class: use synchronous=False if play_next_() is deferred to a callback, timer, or loop(); use synchronous=True if play_next_() always runs before the initial play/play_complex call returns
WARNING register_action('http_request.send', ...) is missing the synchronous= parameter. Defaulting to synchronous=False (safe but prevents StringRef optimization). Check the C++ class: use synchronous=False if play_next_() is deferred to a callback, timer, or loop(); use synchronous=True if play_next_() always runs before the initial play/play_complex call returns
WARNING The minimum WiFi authentication mode (wifi -> min_auth_mode) is not set. This controls the weakest encryption your device will accept when connecting to WiFi. Currently defaults to WPA (less secure), but will change to WPA2 (more secure) in 2026.6.0. WPA uses TKIP encryption which has known security vulnerabilities and should be avoided. WPA2 uses AES encryption which is significantly more secure. To silence this warning, explicitly set min_auth_mode under 'wifi:'. If your router supports WPA2 or WPA3, set 'min_auth_mode: WPA2'. If your router only supports WPA, set 'min_auth_mode: WPA'.
INFO Generating C++ source...
INFO Compiling app... Build path: /data/build/soyo-4
Processing soyo-4 (board: d1_mini; framework: arduino; platform: platformio/espressif8266@4.2.1)
--------------------------------------------------------------------------------
Library Manager: Installing ESP32Async/ESPAsyncTCP @ 2.0.0
Unpacking  [####################################]  100%
Library Manager: ESPAsyncTCP@2.0.0 has been installed!
Library Manager: Installing ESP32Async/ESPAsyncWebServer @ 3.9.6
Unpacking  [####################################]  100%
Library Manager: ESPAsyncWebServer@3.9.6 has been installed!
Library Manager: Resolving dependencies...
Library Manager: Installing heman/AsyncMqttClient-esphome @ 2.0.0
Unpacking  [####################################]  100%
Library Manager: AsyncMqttClient-esphome@2.0.0 has been installed!
Library Manager: Resolving dependencies...
Library Manager: Installing esphome/ESPAsyncTCP-esphome @ ^2.0.0
Unpacking  [####################################]  100%
Library Manager: ESPAsyncTCP-esphome@2.0.0 has been installed!
Library Manager: Installing bblanchon/ArduinoJson @ 7.4.2
Unpacking  [####################################]  100%
Library Manager: ArduinoJson@7.4.2 has been installed!
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
ESPHome: Excluding Updater.cpp from build (using native OTA backend)
ESPHome: Excluding core_esp8266_waveform_phase.cpp from build (waveform not required)
ESPHome: Excluding core_esp8266_waveform_pwm.cpp from build (waveform not required)
Dependency Graph
|-- ESPAsyncTCP @ 2.0.0
|-- ESP8266WiFi @ 1.0
|-- ESPAsyncWebServer @ 3.9.6
|-- ESP8266mDNS @ 1.2
|-- AsyncMqttClient-esphome @ 2.0.0
|-- ArduinoJson @ 7.4.2
|-- ESP8266HTTPClient @ 1.2
Compiling .pioenvs/soyo-4/src/esphome/components/binary_sensor/automation.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/binary_sensor/binary_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/binary_sensor/filter.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/button/button.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/core.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/crash_handler.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/gpio.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/helpers.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/preferences.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/printf_stubs.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esp8266/waveform_stubs.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/esphome/ota/ota_esphome.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/http_request/http_request.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/http_request/http_request_arduino.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/json/json_util.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/logger/logger.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/logger/logger_esp8266.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/md5/md5.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mdns/mdns_component.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mdns/mdns_esp8266.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/custom_mqtt_device.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_alarm_control_panel.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_binary_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_button.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_client.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_climate.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_component.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_cover.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_date.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_datetime.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_event.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_fan.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_light.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_lock.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_number.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_select.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_switch.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_text.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_text_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_time.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_update.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/mqtt/mqtt_valve.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/network/util.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/number/automation.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/number/number.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/number/number_call.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/number/number_traits.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/ota/ota_backend.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/ota/ota_backend_esp8266.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/ota/ota_backend_host.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/select/select.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/select/select_call.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/select/select_traits.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/sensor/automation.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/sensor/sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/sha256/sha256.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/socket/lwip_raw_tcp_impl.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/socket/socket.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_display/button/soyosource_button.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_display/number/soyosource_number.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_display/select/soyosource_select.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_display/soyosource_display.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_modbus/soyosource_modbus.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_virtual_meter/number/soyosource_number.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_virtual_meter/soyosource_virtual_meter.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/soyosource_virtual_meter/switch/soyosource_switch.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/switch/automation.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/switch/switch.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/template/sensor/template_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/text_sensor/filter.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/text_sensor/text_sensor.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/uart/uart.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/uart/uart_component.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/uart/uart_component_esp8266.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/uart/uart_debugger.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/watchdog/watchdog.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/web_server/list_entities.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/web_server/web_server.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/web_server_base/web_server_base.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/wifi/wifi_component.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/components/wifi/wifi_component_esp8266.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/application.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/color.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/component.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/component_iterator.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/controller_registry.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/entity_base.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/gpio.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/helpers.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/log.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/scheduler.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/time.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/time_64.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/util.cpp.o
Compiling .pioenvs/soyo-4/src/esphome/core/wake.cpp.o
Compiling .pioenvs/soyo-4/src/main.cpp.o
Generating LD script .pioenvs/soyo-4/ld/local.eagle.app.v6.common.ld
Compiling .pioenvs/soyo-4/lib1c6/ESPAsyncTCP/AsyncPrinter.cpp.o
Compiling .pioenvs/soyo-4/lib1c6/ESPAsyncTCP/ESPAsyncTCP.cpp.o
Compiling .pioenvs/soyo-4/lib1c6/ESPAsyncTCP/ESPAsyncTCPbuffer.cpp.o
Compiling .pioenvs/soyo-4/lib1c6/ESPAsyncTCP/SyncClient.cpp.o
Compiling .pioenvs/soyo-4/lib1c6/ESPAsyncTCP/tcp_axtls.c.o
Compiling .pioenvs/soyo-4/lib67b/ESP8266WiFi/BearSSLHelpers.cpp.o
Compiling .pioenvs/soyo-4/lib67b/ESP8266WiFi/CertStoreBearSSL.cpp.o
In file included from src/esphome/components/binary_sensor/automation.h:8,
                 from src/esphome.h:5,
                 from src/main.cpp:3:
src/esphome/core/automation.h: In instantiation of 'esphome::TemplatableFn<T, X>::TemplatableFn(V) requires !(invocable<V, X ...>) && !(convertible_to<V, T (*)()>) [with V = const char*; T = const char*; X = {}]':
src/esphome/components/http_request/http_request.h:183:3:   required from 'void esphome::http_request::HttpRequestSendAction<Ts>::set_method(V) [with V = const char*; Ts = {}]'
src/main.cpp:1398:58:   required from here
src/esphome/core/automation.h:70:29: error: static assertion failed: Missing cg.templatable(...) in Python codegen for this TEMPLATABLE_VALUE field. The wrapper was always required; it worked by accident because the old TemplatableValue implicitly converted raw constants. TemplatableFn cannot. See https://developers.esphome.io/blog/2026/04/09/templatablefn-4-byte-templatable-storage-for-trivially-copyable-types/
   70 |     static_assert(sizeof(V) == 0, "Missing cg.templatable(...) in Python codegen for this TEMPLATABLE_VALUE "
      |                   ~~~~~~~~~~^~~~
src/esphome/core/automation.h: In instantiation of 'esphome::TemplatableFn<T, X>::TemplatableFn(V) requires !(invocable<V, X ...>) && !(convertible_to<V, T (*)()>) [with V = bool; T = bool; X = {}]':
src/esphome/components/http_request/http_request.h:186:3:   required from 'void esphome::http_request::HttpRequestSendAction<Ts>::set_capture_response(V) [with V = bool; Ts = {}]'
src/main.cpp:1399:67:   required from here
src/esphome/core/automation.h:70:29: error: static assertion failed: Missing cg.templatable(...) in Python codegen for this TEMPLATABLE_VALUE field. The wrapper was always required; it worked by accident because the old TemplatableValue implicitly converted raw constants. TemplatableFn cannot. See https://developers.esphome.io/blog/2026/04/09/templatablefn-4-byte-templatable-storage-for-trivially-copyable-types/
Archiving .pioenvs/soyo-4/lib1c6/libESPAsyncTCP.a
Indexing .pioenvs/soyo-4/lib1c6/libESPAsyncTCP.a
Compiling .pioenvs/soyo-4/lib67b/ESP8266WiFi/ESP8266WiFi.cpp.o
*** [.pioenvs/soyo-4/src/main.cpp.o] Error 1
========================= [FAILED] Took 103.57 seconds =========================

Und hier das verwendete YAML File:

substitutions:
  name: soyo-4
  device_description: "Monitor and control the Soyosource GTN (WiFi version) via the display port and control the power output on demand via RS485"
  external_components_source: github://syssi/esphome-soyosource-gtn-virtual-meter@main
  # The display port should be connected to these GPIOs using a USB connector (VCC, RX, TX, GND)
  # A optocoupler / optical isolator (f.e. ADUM1201) between the display port and the ESP is recommended if the ESP isn't powered by the display port!
  tx_pin_ttl_wifi: GPIO4
  rx_pin_ttl_wifi: GPIO5
  tx_pin_rs485: GPIO14
  rx_pin_rs485: GPIO12
  tasmota_status_url: "http://192.168.2.56/cm?cmnd=status+10"

esphome:
  name: ${name} 
  comment: ${device_description}
  min_version: 2024.6.0
  project:
    name: "syssi.esphome-soyosource-gtn-virtual-meter"
    version: 2.2.0

esp8266:
  board: d1_mini

external_components:
  - source: ${external_components_source}
    refresh: 0s
  - source: github://pr#7884
    components: [ http_request ]
    
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

ota:
  platform: esphome
  
web_server:

logger:
  level: INFO

# If you use Home Assistant please remove this `mqtt` section and uncomment the `api` component!
# The native API has many advantages over MQTT: https://esphome.io/components/api.html#advantages-over-mqtt
mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_username
  password: !secret mqtt_password
  id: mqtt_client

# api:

uart:
  - id: uart_0
    baud_rate: 9600
    tx_pin: ${tx_pin_ttl_wifi}
    rx_pin: ${rx_pin_ttl_wifi}

  - id: uart_1
    baud_rate: 4800
    tx_pin: ${tx_pin_rs485}
    rx_pin: ${rx_pin_rs485}

soyosource_modbus:
  - id: modbus0
    uart_id: uart_1
    # flow_control_pin: GPIO12

    # Optional settings
    #
    # The name is used as prefix for some log messages and should
    # help to distinguish between different instances/devices
    name: firstfloor_inverter

soyosource_virtual_meter:
  - id: virtualmeter0
    soyosource_modbus_id: modbus0

    # The state of this sensor (instantaneous power in watt) is used as source
    power_id: powermeter0

    # Optional settings
    power_sensor_inactivity_timeout: 20s
    power_demand_calculation: NEGATIVE_MEASUREMENTS_REQUIRED
    min_power_demand: 0
    max_power_demand: 900
    zero_output_on_min_power_demand: true
    # Split/distribute the power demand if you have multiple inverters attached to the same RS485 bus
    power_demand_divider: 1
    # A positive buffer value (10) tries to avoid exporting power to the grid (demand - 10 watts)
    # A negative buffer value (-10) exports power to the grid (demand + 10 watts)
    buffer: 10
    # The operation_status_id sensor is expected here. Passing the operation_status won't work
    # The state is used to suspend the limiter if the operation status of the inverter isn't 0x0 (normal)
    operation_status_id: operation_status_id0

    # The update interval is important and defaults to 3 seconds. If the demand is sent too frequently
    # or rarely the interverter stops. TODO: Identify and validate the lower and upper update limit
    update_interval: 3s

soyosource_display:
  - id: display0
    uart_id: uart_0
    protocol_version: SOYOSOURCE_WIFI_VERSION
    update_interval: 5s

binary_sensor:
  - platform: soyosource_display
    fan_running:
      name: "${name} fan running"
    limiter_connected:
      name: "${name} limiter connected"

button:
  - platform: soyosource_display
    restart:
      name: "${name} restart"

number:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    buffer:
      name: "${name} buffer"
      initial_value: 10
      restore_value: true
    manual_power_demand:
      name: "${name} manual power demand"
      max_value: 1800
    max_power_demand:
      name: "${name} max power demand"
      initial_value: 600
      max_value: 1800
      restore_value: true
    power_demand_divider:
      name: "${name} power demand divider"
      initial_value: 1
      restore_value: true

  - platform: soyosource_display
    start_voltage:
      name: "${name} start voltage"
    shutdown_voltage:
      name: "${name} shutdown voltage"
    # Maximum output power in limiter mode / Output power in constant power mode
    output_power_limit:
      name: "${name} output power limit"
    start_delay:
      name: "${name} start delay"

select:
  - platform: soyosource_display
    operation_mode:
      name: "${name} operation mode"
      optionsmap:
        1: "PV"
        2: "Battery Constant Power"
        17: "PV Limit"
        18: "Battery Limit"
        
http_request:
  id: http_request0
  useragent: esphome/device
  timeout: 1000ms
  verify_ssl: false 

  
sensor:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    power_demand:
      name: "${name} power demand"

  - platform: soyosource_display
    error_bitmask:
      name: "${name} error bitmask"
    operation_mode_id:
      name: "${name} operation mode id"
    operation_status_id:
      name: "${name} operation status id"
      id: operation_status_id0
    battery_voltage:
      name: "${name} battery voltage"
    battery_current:
      name: "${name} battery current"
    battery_power:
      name: "${name} battery power"
    ac_voltage:
      name: "${name} ac voltage"
    ac_frequency:
      name: "${name} ac frequency"
    temperature:
      name: "${name} temperature"
    output_power:
      name: "${name} output power"

  - platform: template
    id: powermeter0
    name: "Instantaneous Power"
    unit_of_measurement: W
    device_class: "power"
    accuracy_decimals: 2
    update_interval: never
   
 

  # mqtt subscribe example
  #- id: powermeter0
  #  internal: true
   # platform: mqtt_subscribe
    #name: "${name} instantaneous power consumption"
    #topic: "smartmeter/sensor/groundfloor/obis/1-0:16.7.0/255/value"
    #accuracy_decimals: 2
    #unit_of_measurement: W
    #device_class: power
    #filters:
     # - throttle_average: 15s

#   # import smartmeter reading from homeassistant
#   # requires the "api" component see above
#   - platform: homeassistant
#     id: powermeter0
#     name: "${name} smartmeter instantaneous power"
#     entity_id: sensor.firstfloor_smartmeter_instantaneous_power
#     filters:
#       - throttle_average: 15s

switch:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    manual_mode:
      name: "${name} manual mode"
      restore_mode: RESTORE_DEFAULT_ON
    emergency_power_off:
      name: "${name} emergency power off"
      restore_mode: RESTORE_DEFAULT_OFF

text_sensor:
  - platform: soyosource_virtual_meter
    soyosource_virtual_meter_id: virtualmeter0
    operation_mode:
      name: "${name} limiter operation mode"

  - platform: soyosource_display
    errors:
      name: "${name} errors"
    operation_mode:
      name: "${name} operation mode"
    operation_status:
      name: "${name} operation status"

interval:
  - interval: 1100ms
    then:
      - http_request.get:
          url: ${tasmota_status_url}
          request_headers:
            Content-Type: application/json
          on_response:
            then:
              - lambda: |-
                  std::string response_data = id(http_request0).get_string();
                  if(!response_data.empty()) {
                    json::parse_json(response_data, [](JsonObject root) {
                      id(powermeter0).publish_state(root["StatusSNS"]["SML"]["Leistung"]);
                    });
                  }

Wie geschrieben, dieses File habe ich seit ein paar Monaten mit Deiner Unterstützung auf 3 ESP`s 8266 am laufen.
Ob Du mir nochmal helfen kannst?

GRuss
Thomas

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -506,6 +506,9 @@ jobs:
           . venv/bin/activate
           echo -e "wifi_ssid: ssid\nwifi_password: password\nmqtt_host: host\nmqtt_username: username\nmqtt_password: password" > tests/secrets.yaml
           esphome -s external_components_source ../components compile tests/esp32c6-compatibility-test.yaml
+          for YAML in tests/esp8266-powermeter-*.yaml; do
+            esphome -s external_components_source ../components compile $YAML
+          done
         env:
           PLATFORMIO_LIBDEPS_DIR: ~/.platformio/libdeps
 

--- a/powermeter-sensor-examples/esphome-http-rest-api.yaml
+++ b/powermeter-sensor-examples/esphome-http-rest-api.yaml
@@ -13,6 +13,7 @@ sensor:
     name: "Instantaneous Power"
     unit_of_measurement: W
     device_class: "power"
+    state_class: measurement
     accuracy_decimals: 2
     update_interval: never
 
@@ -21,14 +22,20 @@ interval:
     then:
       - http_request.get:
           url: ${esphome_rest_api_sensor_url}
+          capture_response: true
+          max_response_buffer_size: 2048
           request_headers:
             Content-Type: application/json
           on_response:
             then:
               - lambda: |-
-                  std::string response_data = id(http_request0).get_string();
-                  if(!response_data.empty()) {
-                    json::parse_json(response_data, [](JsonObject root) {
-                      id(powermeter0).publish_state(root["value"]);
+                  if (response->status_code == 200 && !body.empty()) {
+                    json::parse_json(body, [](JsonObject root) -> bool {
+                      auto value = root["value"];
+                      if (!value.isNull()) {
+                        id(powermeter0).publish_state(value.as<float>());
+                        return true;
+                      }
+                      return false;
                     });
                   }

--- a/powermeter-sensor-examples/mqtt-topic.yaml
+++ b/powermeter-sensor-examples/mqtt-topic.yaml
@@ -12,5 +12,6 @@ sensor:
     accuracy_decimals: 2
     unit_of_measurement: W
     device_class: power
+    state_class: measurement
     filters:
       - throttle_average: 15s

--- a/powermeter-sensor-examples/shelly-3em-http-status-json.yaml
+++ b/powermeter-sensor-examples/shelly-3em-http-status-json.yaml
@@ -13,6 +13,7 @@ sensor:
     name: "Instantaneous Power"
     unit_of_measurement: W
     device_class: "power"
+    state_class: measurement
     accuracy_decimals: 2
     update_interval: never
 
@@ -28,7 +29,13 @@ interval:
           on_response:
             then:
               - lambda: |-
-                  json::parse_json(body, [](JsonObject root) -> bool {
-                    id(powermeter0).publish_state(root["total_power"]);
-                    return true;
-                  });
+                  if (response->status_code == 200 && !body.empty()) {
+                    json::parse_json(body, [](JsonObject root) -> bool {
+                      auto value = root["total_power"];
+                      if (!value.isNull()) {
+                        id(powermeter0).publish_state(value.as<float>());
+                        return true;
+                      }
+                      return false;
+                    });
+                  }

--- a/powermeter-sensor-examples/tasmota-http-status-sns-en.yaml
+++ b/powermeter-sensor-examples/tasmota-http-status-sns-en.yaml
@@ -15,6 +15,7 @@ sensor:
     name: "Instantaneous Power"
     unit_of_measurement: W
     device_class: "power"
+    state_class: measurement
     accuracy_decimals: 2
     update_interval: never
 
@@ -23,14 +24,20 @@ interval:
     then:
       - http_request.get:
           url: ${tasmota_status_url}
+          capture_response: true
+          max_response_buffer_size: 2048
           request_headers:
             Content-Type: application/json
           on_response:
             then:
               - lambda: |-
-                  std::string response_data = id(http_request0).get_string();
-                  if(!response_data.empty()) {
-                    json::parse_json(response_data, [](JsonObject root) {
-                      id(powermeter0).publish_state(root["StatusSNS"]["SML"]["curr_w"]);
+                  if (response->status_code == 200 && !body.empty()) {
+                    json::parse_json(body, [](JsonObject root) -> bool {
+                      auto value = root["StatusSNS"]["SML"]["curr_w"];
+                      if (!value.isNull()) {
+                        id(powermeter0).publish_state(value.as<float>());
+                        return true;
+                      }
+                      return false;
                     });
                   }

--- a/powermeter-sensor-examples/tasmota-http-status-sns.yaml
+++ b/powermeter-sensor-examples/tasmota-http-status-sns.yaml
@@ -15,6 +15,7 @@ sensor:
     name: "Instantaneous Power"
     unit_of_measurement: W
     device_class: "power"
+    state_class: measurement
     accuracy_decimals: 2
     update_interval: never
 
@@ -23,14 +24,20 @@ interval:
     then:
       - http_request.get:
           url: ${tasmota_status_url}
+          capture_response: true
+          max_response_buffer_size: 2048
           request_headers:
             Content-Type: application/json
           on_response:
             then:
               - lambda: |-
-                  std::string response_data = id(http_request0).get_string();
-                  if(!response_data.empty()) {
-                    json::parse_json(response_data, [](JsonObject root) {
-                      id(powermeter0).publish_state(root["StatusSNS"]["SML"]["Leistung"]);
+                  if (response->status_code == 200 && !body.empty()) {
+                    json::parse_json(body, [](JsonObject root) -> bool {
+                      auto value = root["StatusSNS"]["SML"]["Leistung"];
+                      if (!value.isNull()) {
+                        id(powermeter0).publish_state(value.as<float>());
+                        return true;
+                      }
+                      return false;
                     });
                   }

